File: DataStructure/2.QueueAndDeque/Group3/minyeong/10866.cpp
#include <iostream>
#include <string>
#include <deque>
using namespace std;

int main(void)
{   
    ios::sync_with_stdio(false); 
    cin.tie(NULL);
    int n;
    cin >> n;
    deque<int>d;
    for(int i=0; i < n; i++)
    {   
        
        string s;
        int input;

        cin >> s;
        if(s == "push_front") 
        {     
            cin >> input;
            d.push_front(input);
        }
        else if(s == "push_back") 
        {     
            cin >> input;
            d.push_back(input);
        }
        else if(s == "pop_front") 
        {   
            if(d.empty()) cout << -1 << "\n";
            else 
            {   
                cout << d.front() << "\n";
                d.pop_front();
            }
        }
         else if(s == "pop_back") 
        {   
            if(d.empty()) cout << -1 << "\n";
            else 
            {   
                cout << d.back() << "\n";
                d.pop_back();
            }
        }
        else if (s == "size") cout << d.size() << "\n";
        else if (s == "empty")
    {
        if (d.empty()) cout << 1 << "\n";
        else cout << 0 << "\n";
    }
        else if (s == "front")
        {
            if(d.empty()) cout << -1 << "\n";
            else cout << d.front() << "\n";
        }
        else if(s == "back")
        {   
            if(d.empty()) cout << -1 << "\n";
            else cout << d.back() << "\n";
        }
    }
   
    
    return 0;
}

File: DataStructure/2.QueueAndDeque/Group3/dooyeon/10866.py
from collections import deque
import sys
input = sys.stdin.readline

N = int(input())
deq = deque([])

for _ in range(N):
    st = input().split()
    if "push_front" == st[0]:
        deq.appendleft(st[1])
    elif "push_back" == st[0]:
        deq.append(st[1])
    elif "pop_front" == st[0]:
        if deq:
            print(deq.popleft())
        else:
            print(-1)
    elif "pop_back" == st[0]:
        if deq:
            print(deq.pop())
        else:
            print(-1)
    elif "size" == st[0]:
        print(len(deq))
    elif "empty" == st[0]:
        if deq:
            print(0)
        else:
            print(1)
    elif "front" == st[0]:
        if deq:
            print(deq[0])
        else:
            print(-1)
    else:
        if deq:
            print(deq[-1])
        else:
            print(-1)